File: backend/src/main/java/com/commercehub/backend/payment/service/PaymentService.java
package com.commercehub.backend.payment.service;

import com.commercehub.backend.payment.dto.request.PaymentRequest;
import com.commercehub.backend.payment.dto.response.PaymentResponse;
import com.commercehub.backend.payment.enums.PaymentStatus;

import java.util.List;

public interface PaymentService {

    /**
     * Create a new payment for an order.
     */
    PaymentResponse createPayment(PaymentRequest request);

    /**
     * Get payment by ID.
     */
    PaymentResponse getPaymentById(Long paymentId);

    /**
     * Get all payment attempts for an order.
     */
    List<PaymentResponse> getPaymentsByOrderId(Long orderId);

    /**
     * Mark payment as successful.
     */
    PaymentResponse markPaymentSuccess(Long paymentId, String gatewayReferenceId);

    /**
     * Mark payment as failed.
     */
    PaymentResponse markPaymentFailed(Long paymentId, String errorMessage);

    /**
     * Cancel a pending payment.
     */
    PaymentResponse cancelPayment(Long paymentId);

    /**
     * Refund a successful payment.
     */
    PaymentResponse refundPayment(Long paymentId);

    /**
     * Retry a failed payment.
     */
    PaymentResponse retryPayment(Long paymentId);

    /**
     * Get payment by transaction ID.
     */
    PaymentResponse getPaymentByTransactionId(String transactionId);

    /**
     * Get all payments having the given status.
     */
    List<PaymentResponse> getPaymentsByStatus(PaymentStatus status);

    /**
     * Get all payments.
     */
    List<PaymentResponse> getAllPayments();
}

File: frontend/src/services/payment.service.ts
import apiClient from '../core/api/axios';
import { API_ENDPOINTS } from '../core/api/apiEndpoints';

export const paymentService = {
  createPayment: (payload: Record<string, any>) =>
    apiClient.post(API_ENDPOINTS.PAYMENTS.BASE, payload),

  getPayments: () => apiClient.get(API_ENDPOINTS.PAYMENTS.BASE),

  getPaymentById: (paymentId: string) =>
    apiClient.get(API_ENDPOINTS.PAYMENTS.BY_ID(paymentId)),

  getPaymentsByOrder: (orderId: string) =>
    apiClient.get(API_ENDPOINTS.PAYMENTS.BY_ORDER(orderId)),

  getPaymentsByStatus: (status: string) =>
    apiClient.get(API_ENDPOINTS.PAYMENTS.BY_STATUS(status)),

  getPaymentByTransaction: (transactionId: string) =>
    apiClient.get(API_ENDPOINTS.PAYMENTS.BY_TRANSACTION(transactionId)),

  markSuccess: (paymentId: string) =>
    apiClient.put(API_ENDPOINTS.PAYMENTS.SUCCESS(paymentId)),

  markFailed: (paymentId: string) =>
    apiClient.put(API_ENDPOINTS.PAYMENTS.FAILED(paymentId)),

  cancelPayment: (paymentId: string) =>
    apiClient.put(API_ENDPOINTS.PAYMENTS.CANCEL(paymentId)),

  refundPayment: (paymentId: string) =>
    apiClient.put(API_ENDPOINTS.PAYMENTS.REFUND(paymentId)),

  retryPayment: (paymentId: string) =>
    apiClient.put(API_ENDPOINTS.PAYMENTS.RETRY(paymentId)),
};
